### PR TITLE
product_module [API: addProduct, updateProduct, deleteProduct, delete…

### PR DESCRIPTION
**Pull request by vidhirathod29**
…ProductImage, listOfProduct]

### CODE REVIEW

File: controller/productController.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const {
+  addProduct,
+  updateProduct,
+  deleteProduct,
+  deleteProductImage,
+  productList,
+} = require('../services/productService');
+
+module.exports = {
+  addProduct: async (req, res, next) => {
```
Comment (ShrutiSachpara) [inline controller/productController.js]: remove async await

File: models/db.js [hunk 2]
```diff
@@ -19,7 +25,22 @@ sequelize
 const db = {};
 db.Sequelize = Sequelize;
 db.sequelize = sequelize;
+db.categoryModel = require('../models/category')(sequelize, Sequelize);
 db.authModel = require('../models/auth')(sequelize, Sequelize);
+db.productModel = require('../models/product')(sequelize, Sequelize);
```
Comment (ShrutiSachpara) [inline models/db.js:34]: add relation module wise another module relation don't add in product module

File: services/productService.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        undefined,
+        RESPONSE_STATUS.SUCCESS,
+      ),
+    );
+  }
+};
+
+const updateProduct = async (req, res, next) => {
+  const id = req.params.id;
+  const findId = await product.findOne({ where: { id } });
```
Comment (ShrutiSachpara) [inline services/productService.js]: find with is_deleted

File: services/productService.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+  const updateProductData = {
+    product_name,
+    price,
+    product_description,
+    product_quantity,
+    images,
+  };
+
+  if (findId) {
+    const updateProduct = await product.update(updateProductData, {
```
Comment (ShrutiSachpara) [inline services/productService.js]: same as here

File: services/productService.js [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        undefined,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  }
+};
+
+const deleteProduct = async (req, res, next) => {
+  const id = req.params.id;
+  const findId = product.findOne({ where: { id } });
```
Comment (ShrutiSachpara) [inline services/productService.js]: same as here

File: services/productService.js [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+        undefined,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  }
+};
+
+const deleteProductImage = async (req, res, next) => {
+  const id = req.params.id;
+  const findId = productImage.findOne({ where: { id } });
```
Comment (ShrutiSachpara) [inline services/productService.js]: same as here

File: utils/enum.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ module.exports = {
     CUSTOMER: 'Customer',
   },
   GENDER: {
-    MALE: 'Male',
     FEMALE: 'Female',
+    MALE: 'Male',
```
Comment (ShrutiSachpara) [inline utils/enum.js:11]: why old code is removed???

File: validation/validator.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-const validator = require('express-joi-validation').createValidator({
+ const validator = require('express-joi-validation').createValidator({
```
Comment (ShrutiSachpara) [inline validation/validator.js]: why chnage here?

File: services/productService.js (hunk 6 above)
Comment (vidhirathod29) [reply services/productService.js]: in product_image table there no is_deleted field that's why I have not done that

**Pull request merged**